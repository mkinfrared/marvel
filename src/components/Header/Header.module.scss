.Header {
  position: sticky;
  top: 0;
  z-index: 1;

  .searchBox {
    margin-left: auto;
  }
}
