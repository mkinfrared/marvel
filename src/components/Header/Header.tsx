import React from "react";
import { Link } from "react-router-dom";

import HeroSearch from "components/HeroSearch";
import logo from "icons/MarvelLogo/icon.svg";
import { Pages } from "routes/Routes.type";
import AppBar from "ui/AppBar";

import css from "./Header.module.scss";

const Header = () => {
  return (
    <header className={css.Header}>
      <AppBar>
        <Link to={Pages.HOME}>
          <img src={logo} alt="logo" />
        </Link>
        <HeroSearch className={css.searchBox} />
      </AppBar>
    </header>
  );
};

export default React.memo(Header);
