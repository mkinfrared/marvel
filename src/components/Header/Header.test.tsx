import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import Header from "./Header";

describe("<Header />", () => {
  const Component = (
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
