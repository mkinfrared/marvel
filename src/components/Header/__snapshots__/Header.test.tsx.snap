// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> match the snapshot 1`] = `
<div>
  <header
    class="Header"
  >
    <div
      class="AppBar"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          src="icon.svg"
        />
      </a>
      <div
        class="Search searchBox"
      >
        <div
          class="Input"
        >
          <input
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
